frontend/src/components/nodes/NodeNav.tsx

```diff
@@ -0,0 +1,136 @@
+import Link from "next/link";
+
+import React from "react";
+import { Row, Col } from "react-bootstrap";
+
+import NodesApi, * as N from "../../libraries/explorer-wamp/nodes";
+
+interface State {
+  nodeStats?: N.NodeStats;
+}
+
+export default class extends React.Component<State> {
+  constructor(props: any) {
+    super(props);
+    this.timer = null;
+  }
+
+  timer: ReturnType<typeof setTimeout> | null;
+  state: State = {};
+
+  componentDidMount() {
+    this.timer = setTimeout(this.regularFetchInfo, 0);
+  }
+
+  componentWillUnmount() {
+    const timer = this.timer;
+    this.timer = null;
+    if (timer !== null) {
+      clearTimeout(timer);
+    }
+  }
+
+  regularFetchInfo = async () => {
+    await this.fetchNodes();
+    if (this.timer !== null) {
+      this.timer = setTimeout(this.regularFetchInfo, 10000);
+    }
+  };
+
+  fetchNodes = async () => {
+    new NodesApi()
+      .getOnlineNodesStats()
+      .then((nodeStats) => this.setState({ nodeStats }));
+  };
+
+  changeValidatorNodeBorder = () => {
+    const validatorDiv = document.getElementById("validator-node");
+    if (validatorDiv) {
+      validatorDiv.style.border = "2px solid #0066ff";
+    }
+    const nonValidatorDiv = document.getElementById("non-validator-node");
+    if (nonValidatorDiv) {
+      nonValidatorDiv.style.border = "2px solid #e6e6e6";
+    }
+  };
+
+  changeNonValidatorNodeBorder = () => {
+    const validatorDiv = document.getElementById("validator-node");
+    if (validatorDiv) {
+      validatorDiv.style.border = "2px solid #e6e6e6";
+    }
+    const nonValidatorDiv = document.getElementById("non-validator-node");
+    if (nonValidatorDiv) {
+      nonValidatorDiv.style.border = "2px solid #0066ff";
+    }
+  };
```

Comment:
It can be done muuuuch simpler by just settings `selected` CSS class on the relevant button based on the `props.role === 'validating'`

Reply:
okay